File: agora-easycall-ios-oc/AgoraSample/Login/AGSLoginViewController.m
//
//  AGSKeyLoginViewController.m
//  AgoraSample
//
//  Created by apple on 15/9/15.
//  Copyright (c) 2015年 Agora. All rights reserved.
//

#import "AGSLoginViewController.h"
#import "AGSSelectMediaViewController.h"

@interface AGSLoginViewController ()

@property (weak, nonatomic) IBOutlet UITextField *keyTextField;
@property (weak, nonatomic) IBOutlet UITextField *usernameTextField;
@property (weak, nonatomic) IBOutlet UIButton *backButton;
@property (weak, nonatomic) IBOutlet UILabel *agoraLabel;
@property (weak, nonatomic) IBOutlet UIButton *loginButton;

@end

@implementation AGSLoginViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    NSString *vendorKey = [[NSUserDefaults standardUserDefaults] stringForKey:AGSKeyVendorKey];
    self.usernameTextField.text = [[NSUserDefaults standardUserDefaults] stringForKey:AGSKeyUsername];
    
    //
    [self.backButton setTitle:NSLocalizedString(@"Return", @"") forState:UIControlStateNormal];
    [self.loginButton setTitle:NSLocalizedString(@"Login", @"") forState:UIControlStateNormal];
    self.agoraLabel.text = NSLocalizedString(@"Agora Easy Call", @"");
    self.keyTextField.placeholder = NSLocalizedString(@"Enter Vendor Key", @"");
    self.usernameTextField.placeholder = NSLocalizedString(@"User", @"");
    
    if (vendorKey) {
        self.keyTextField.text = vendorKey;
    } else {
        NSError *err = nil;
        NSURL *innerKeyUrl = [NSURL URLWithString:@"http://192.168.99.253:8970/agora.inner.test.key.txt"];
        NSString *innerVendorKey = [NSString stringWithContentsOfURL:innerKeyUrl
                                                            encoding:NSASCIIStringEncoding
                                                               error:&err];

        if (!err) {
            self.keyTextField.text = [innerVendorKey
                                      stringByReplacingOccurrencesOfString:@"\n" withString:@""];; // Please use your own key. The inner test key is just invalid in public.
        }
    }
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)isValidateInput
{
    [self.view endEditing:YES];
    if (self.keyTextField.text.length == 0) {
        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil
                                                            message:NSLocalizedString(@"Vendor key can not be empty", @"")
                                                           delegate:nil
                                                  cancelButtonTitle:NSLocalizedString(@"Done", @"")
                                                  otherButtonTitles:nil];
        [alertView show];
        return NO;
    }
    if (self.usernameTextField.text.length == 0) {
        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil
                                                            message:NSLocalizedString(@"User name can not be empty", @"")
                                                           delegate:nil
                                                  cancelButtonTitle:NSLocalizedString(@"Done", @"")
                                                  otherButtonTitles:nil];
        [alertView show];
        return NO;
    }
    
    return YES;
}

- (IBAction)didClickEnterChatRoom:(id)sender
{
    if ([self isValidateInput]) {
        [self performSegueWithIdentifier:AGSKeySegueIdentifierSelect sender:self];
        
        NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
        [userDefaults setObject:self.keyTextField.text forKey:AGSKeyVendorKey];
        [userDefaults setObject:self.usernameTextField.text forKey:AGSKeyUsername];
    }
}



@end

File: agora-easycall-ios-swift/SampleSwift/AGSLoginViewController.swift
//
//  AGSLoginViewController.swift
//  SampleSwift
//
//  Created by let on 10/19/15.
//  Copyright © 2015 Agora. All rights reserved.
//

import UIKit
import Foundation

class AGSLoginViewController: AGSBaseViewController { //:base

    // Outlet
    
    @IBOutlet weak var keyTextField: UITextField!
    @IBOutlet weak var usernameTextField: UITextField!
    @IBOutlet weak var backButton: UIButton!
    @IBOutlet weak var agoraLabel: UILabel!
    @IBOutlet weak var loginButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
//        keyTextField.text = NSUserDefaults.standardUserDefaults().stringForKey(AGSKeys.vendorKey)
        usernameTextField.text = NSUserDefaults.standardUserDefaults().stringForKey(AGSKeys.username)
        
        //
        backButton.setTitle(NSLocalizedString("Return", comment: ""), forState: UIControlState.Normal)
        loginButton.setTitle(NSLocalizedString("Login", comment: ""), forState: UIControlState.Normal)
        agoraLabel.text = NSLocalizedString("Agora Easy Call", comment: "")
        keyTextField.placeholder = NSLocalizedString("Enter Vendor Key", comment: "");
        usernameTextField.placeholder = NSLocalizedString("User", comment: "");
        
        let vendorKey = NSUserDefaults.standardUserDefaults().stringForKey(AGSKeys.vendorKey)
        if vendorKey != nil {
            keyTextField.text = vendorKey
        } else {
            let innerKeyUrl = NSURL(string: "http://192.168.99.253:8970/agora.inner.test.key.txt")
            if let jsonData: NSData = NSData(contentsOfURL: innerKeyUrl!) {
                let innerVendorKey = NSString(data: jsonData, encoding: NSUTF8StringEncoding)
                self.keyTextField.text = innerVendorKey!.stringByReplacingOccurrencesOfString("\n", withString: "") // Please use your own key. The inner test key is just invalid in public.
            }

        }
     
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func isValidateInput() -> Bool {
        view.endEditing(true)
        
        if keyTextField.text?.characters.count == 0 {
            let alert = UIAlertView(title: "", message: NSLocalizedString("Vendor key can not be empty", comment: ""), delegate: nil, cancelButtonTitle: NSLocalizedString("Done", comment: ""))
            alert.show()
            return false;
        }
        if usernameTextField.text?.characters.count == 0 {
            let alert = UIAlertView(title: "", message: NSLocalizedString("User name can not be empty", comment: ""), delegate: nil, cancelButtonTitle: NSLocalizedString("Done", comment: ""))
            alert.show()
            return false;
        }
        return true
    }
    
    @IBAction func didClickEnterChatRoom(sender: AnyObject) {
        if isValidateInput() {
            performSegueWithIdentifier(AGSSegueIdentifiers.select, sender: self)
            let defualts = NSUserDefaults.standardUserDefaults()
            defualts.setObject(keyTextField.text, forKey: AGSKeys.vendorKey)
            defualts.setObject(usernameTextField.text, forKey: AGSKeys.username)
        }
    }
    
}
